core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/StructuredEventFilter.java

```diff
@@ -196,11 +188,25 @@ private void sendStructuredEvent(RestRequestDetails restRequest, RestResponseDet
     }
 
     private void extendRestParamsFromResponse(Map<String, String> params, CharSequence responseBody) {
-        if (!params.containsKey(RESOURCE_ID)) {
-            Pattern pattern = Pattern.compile("\"id\":([0-9]*)");
-            Matcher matcher = pattern.matcher(responseBody);
-            if (matcher.find() && matcher.groupCount() >= 1) {
-                params.put(RESOURCE_ID, matcher.group(1));
+        if (responseBody != null && !params.containsKey(RESOURCE_ID)) {
+            String resourceId = null;
+            try {
+                JsonNode jsonNode = JsonUtil.readTree(responseBody.toString());
+                JsonNode idNode = jsonNode.path("id");
+                resourceId = idNode.asText();
+            } catch (IOException e) {
+                LOGGER.warn("Parsing of ID from JSON response failed", e);
+            }
+            if (resourceId == null || resourceId.isEmpty()) {
+                Pattern pattern = Pattern.compile("\"id\":([0-9]*)");
```

Comment:
Pattern compile is heavy operation would be better to do it once

Reply:
ok